parser/src/parser/tests/mod.rs

```diff
@@ -1,18 +1,619 @@
+            AccessType::Index($idx),
+            0,
+        ))
+    };
+
+    ($name:literal [ $idx:literal ]) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Index($idx),
+            0,
+        ))
+    };
+
+    ($name:ident [ $row:literal ] [ $col:literal ]) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Matrix($row, $col),
+            0,
+        ))
+    };
+
+    ($name:ident [ $row:literal ] [ $col:literal ], $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Matrix($row, $col),
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+
+    ($module:ident, $name:ident [ $idx:literal ], $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ident!($module, $name).into(),
+            access_type: AccessType::Index($idx),
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:ident [ $idx:literal ], $offset:literal) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Index($idx),
+            $offset,
+        ))
+    };
+
+    ($name:ident [ $idx:literal ], $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Index($idx),
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:ident [ $idx:literal ], $offset:literal, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Index($idx),
+            offset: $offset,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:literal [ $idx:literal ], $offset:literal) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Index($idx),
+            $offset,
+        ))
+    };
+
+    ($module:ident, $name:ident [ $idx:literal ], $offset:literal, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Resolved(ident!($module, $name)),
+            access_type: AccessType::Index($idx),
+            offset: $offset,
+            ty: Some($ty),
+        })
+    };
```

Comment:
This seems like another case that shouldn't come up, since trace columns are only defined in the root module